packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/PersistedNameTests.kt

```diff
@@ -232,9 +232,7 @@ class PersistedNameTests {
 
         assertNotNull(dynamicSample)
         assertEquals("Realm", dynamicSample.getValue("persistedNameStringField"))
-        assertFailsWithMessage<IllegalArgumentException>("Schema for type 'AlternativePersistedNameSample' doesn't contain a property named 'publicNameStringField'") {
-            dynamicSample.getValue("publicNameStringField")
-        }
+        assertEquals("Realm", dynamicSample.getValue("publicNameStringField"))
```

Comment:
It would probably be appropriate with a comment here that this is only due to the native of `asDynamicRealm` that the public name is available here.